ntbs-service/wwwroot/source/Components/ValidatePostcode.ts

```diff
@@ -0,0 +1,43 @@
+import Vue from 'vue';
+import { getHeaders } from '../helpers';
+import axios from 'axios';
+
+const ValidatePostcode = Vue.extend({
+    props: ['shouldvalidatefull'],
+    methods: {
+      validate: function (event: FocusEvent) {
+          // Our onBlur validate events happen on input fields
+          const inputField = event.target as HTMLInputElement
+          const newValue = inputField.value;
+
+        let requestConfig = {
+          url: `${window.location.pathname}/ValidatePostcode`,
+          headers: getHeaders(),
+          params: {
+            "shouldValidateFull": this.$props.shouldvalidatefull,
+            "postcode": newValue
+          }
+        }
+        axios.request(requestConfig)
+        .then((response: any) => {
+            var errorMessage = response.data;
+            var hasError = errorMessage != '';

```

Comment:
`errorMessage != ''` is essentially the same as `!errorMessage` but reads as if it isn't
If we're checking if it has actually returned an empty string try `!== ''`, or if we're just checking if it's empty/null try `!errorMessage`

Reply:
If just checking for truthiness, could be inlined into `if (hasError)` too